src/palimpzest/query/operators/logical.py

```diff
@@ -381,33 +381,79 @@ def get_logical_op_params(self) -> dict:
 class GroupByAggregate(LogicalOperator):
     def __init__(
         self,
-        group_by_sig: GroupBySig,
+        group_by_sig: GroupBySig | None = None,
+        gby_fields: list[str] | None = None,
+        agg_fields: list[str] | None = None,
+        agg_funcs: list[str] | None = None,
         *args,
         **kwargs,
     ):
         super().__init__(*args, **kwargs)
         if not self.input_schema:
             raise ValueError("GroupByAggregate requires an input schema")
-        (valid, error) = group_by_sig.validate_schema(self.input_schema)
-        if not valid:
-            raise TypeError(error)
-        self.group_by_sig = group_by_sig
+        
+        # Support both old GroupBySig and new direct parameters
+        if group_by_sig is not None:
+            # Old API: using GroupBySig
+            (valid, error) = group_by_sig.validate_schema(self.input_schema)
+            if not valid:
+                raise TypeError(error)
+            self.group_by_sig = group_by_sig
+            self.gby_fields = group_by_sig.group_by_fields
+            self.agg_fields = group_by_sig.agg_fields
+            self.agg_funcs = group_by_sig.agg_funcs
+        else:
+            # New API: using direct parameters
+            if gby_fields is None or agg_fields is None or agg_funcs is None:
+                raise ValueError("Must provide either group_by_sig or all of (gby_fields, agg_fields, agg_funcs)")
+            
+            # Validate fields exist in input schema
+            for f in gby_fields:
+                if f not in self.input_schema.model_fields:
+                    raise TypeError(f"Supplied schema has no field {f}")
```

Comment:
We can drop this check as semantic group by operations can group over fields which do not exist in the input schema. In our example where we are using a semantic group by to group complaints from customer reviews, the `complaint` field does not yet exist in the input schema. (The `contents` field does exist however, and we must have the `agg_fields` present in the input schema; hence the second check should remain).